ui-manchette/src/components/Manchette.tsx

```diff
@@ -26,40 +29,82 @@ const Manchette = ({
   resetZoom,
   yZoom = 1,
   waypoints,
-  activeWaypointId,
+  waypointMenuData,
   isProportional = true,
   toggleMode,
   children,
   height = INITIAL_OP_LIST_HEIGHT,
-}: ManchetteProps) => (
-  <div className="manchette-container">
-    <div className="bg-white-100 border-r border-grey-30" style={{ height: `${height}px` }}>
-      <WaypointList waypoints={waypoints} activeWaypointId={activeWaypointId} />
-      {children}
-    </div>
-    <div className="manchette-actions">
-      <div className="zoom-buttons">
-        <button className="zoom-out" onClick={zoomYOut} disabled={yZoom <= MIN_ZOOM_Y}>
-          <ZoomOut />
-        </button>
-        <button className="zoom-in" onClick={zoomYIn} disabled={yZoom >= MAX_ZOOM_Y}>
-          <ZoomIn />
-        </button>
-        <button className="zoom-reset" onClick={resetZoom}>
-          Fit
-        </button>
+}: ManchetteProps) => {
+  const [menuPosition, setMenuPosition] = useState<number>();
+
+  // Allow to track the menu position after we might have scrolled in the page
+  useEffect(() => {
+    const manchetteWrapperPosition =
+      waypointMenuData?.manchetteWrapperRef?.current?.getBoundingClientRect().top;
+    const waypointPosition =
+      waypointMenuData?.activeWaypointRef?.current?.getBoundingClientRect().top;
+
+    if (!manchetteWrapperPosition || !waypointPosition) {
+      setMenuPosition(undefined);
+    } else {
+      setMenuPosition(window.scrollY * 2 + manchetteWrapperPosition + waypointPosition);
```

Comment:
Hm, not sure how this works, and seems like this doesn't work on my machine… Why multiply the window scroll offset by 2?

The menu has `position: absolute`, and it's inside a wrapper which has `position: relative`. This means we need to compute the distance between these two elements. `waypointPosition - manchetteWrapperPosition` worked well for me.

Reply:
Oh right, I was so focused on trying to handle the scrollY that I didn't think to do it that way.
Done but had to take the bottom `getBoundingClientRect` for the waypoint position.